src/rpc/ckb.rs

```diff
@@ -81,3 +83,152 @@ crate::jsonrpc!(pub struct CkbRpcClient {
     pub fn update_main_logger(&mut self, config: MainLoggerConfig) -> ();
     pub fn set_extra_logger(&mut self, name: String, config_opt: Option<ExtraLoggerConfig>) -> ();
 });
+
+fn transform_cycles(cycles: Option<Vec<ckb_jsonrpc_types::Cycle>>) -> Vec<Cycle> {
+    if let Some(cycles) = cycles {
+        cycles.into_iter().map(|c| c.into()).collect()
+    } else {
+        vec![]
+    }
+}
+
+impl CkbRpcClient {
+    pub fn get_packed_block(&mut self, hash: H256) -> Result<Option<JsonBytes>, crate::RpcError> {
+        self.post("get_block", (hash, Some(Uint32::from(0u32))))
+    }
+
+    // turn block response into BlcokView and cycle vec
+    fn blockresponse2view(
+        opt_resp: Option<BlockResponse>,
+    ) -> Result<Option<(BlockView, Vec<Cycle>)>, crate::rpc::RpcError> {
+        if let Some(resp) = opt_resp {
+            match resp {
+                BlockResponse::Regular(block_view) => Ok(Some((block_view.get_value()?, vec![]))),
+                BlockResponse::WithCycles(block_cycles) => {
+                    let block_view = block_cycles.block.get_value()?;
+                    let cycles = transform_cycles(block_cycles.cycles);
+
+                    Ok(Some((block_view, cycles)))
+                }
+            }
+        } else {
+            Ok(None)
+        }
```

Comment:
prefer transpose to if let match, it's more rust idioms

```suggestion
        opt_resp
            .map(|resp| match resp {
                BlockResponse::Regular(block_view) => Ok((block_view.get_value()?, vec![])),
                BlockResponse::WithCycles(block_cycles) => {
                    let cycles = transform_cycles(block_cycles.cycles);
                    Ok((block_cycles.block.get_value()?, cycles))
                }
            })
            .transpose()
```